src/components/image/index.js

```diff
@@ -49,7 +49,11 @@ class Image extends PureBaseComponent {
      * The type of overly to place on top of the image. Note: the image MUST have proper size, see examples in:
      * https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/OverlaysScreen.js
      */
-    overlayType: Overlay.propTypes.type
+    overlayType: Overlay.propTypes.type,
+    /**
+     * Render an overlay with custom content
+     */
+    customOverlay: PropTypes.func
```

Comment:
Consider renaming to `customOverlayContent` cause `customOverlay` sounds like you can provide custom gradient overlay